packages/next/src/server/web/spec-extension/unstable-cache.ts

```diff
@@ -129,6 +125,10 @@ export function unstable_cache<T extends Callback>(
       // router)  and if there's no static generation store, we aren't in app
       // router. Default to an empty pathname and search params when there's no
       // request store or static generation store available.
+      const requestStore =
+        workUnitStore && workUnitStore.type === 'request'
+          ? workUnitStore
+          : undefined
```

Comment:
Similar to the patch-fetch comment I think this might sometimes hide pathname from cache keys during prerender that used to be present

Reply:
Rebased on #70889.